File: SYP.Web/Modules/Order/TieredDiscountSettings/TieredDiscountSettingsColumns.cs
namespace SYP.Order.Columns;

[ColumnsScript("Order.TieredDiscountSettings")]
[BasedOnRow(typeof(TieredDiscountSettingsRow), CheckNames = true)]
public class TieredDiscountSettingsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public decimal MinAmount { get; set; }
    public decimal DiscountPercentage { get; set; }
    public bool IsActive { get; set; }
    public int DisplayOrder { get; set; }
    public DateTime InsertDate { get; set; }
    public int InsertUserId { get; set; }
    public DateTime UpdateDate { get; set; }
    public int UpdateUserId { get; set; }
}

File: SYP.Web/Modules/ServerTypes/Order/TieredDiscountSettingsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { TieredDiscountSettingsRow } from "./TieredDiscountSettingsRow";

export interface TieredDiscountSettingsColumns {
    Id: Column<TieredDiscountSettingsRow>;
    MinAmount: Column<TieredDiscountSettingsRow>;
    DiscountPercentage: Column<TieredDiscountSettingsRow>;
    IsActive: Column<TieredDiscountSettingsRow>;
    DisplayOrder: Column<TieredDiscountSettingsRow>;
    InsertDate: Column<TieredDiscountSettingsRow>;
    InsertUserId: Column<TieredDiscountSettingsRow>;
    UpdateDate: Column<TieredDiscountSettingsRow>;
    UpdateUserId: Column<TieredDiscountSettingsRow>;
}

export class TieredDiscountSettingsColumns extends ColumnsBase<TieredDiscountSettingsRow> {
    static readonly columnsKey = 'Order.TieredDiscountSettings';
    static readonly Fields = fieldsProxy<TieredDiscountSettingsColumns>();
}